sdk/src/main/java/io/opentelemetry/sdk/errorhandler/DefaultErrorHandler.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.errorhandler;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Default implementation of {@link ErrorHandler}. By default, logs the full string of the error
+ * with a level of "Warning".
+ *
+ * @since 0.1.0
+ */
+@ThreadSafe
+public final class DefaultErrorHandler implements ErrorHandler {
+  private static final Logger logger = Logger.getLogger(DefaultErrorHandler.class.getName());
+  private static final ErrorHandler instance = new DefaultErrorHandler();
+
+  /**
+   * Returns a {@code ErrorHandler} singleton that is the default implementation for {@link
+   * ErrorHandler}.
+   *
+   * @return a {@code ErrorHandler} singleton that is the default implementation for {@link
+   *     ErrorHandler}.
+   */
+  public static ErrorHandler getInstance() {
+    return instance;
+  }
+
+  @Override
+  public void handle(OpenTelemetryException e) {
```

Comment:
Just curious do we have a story about controlling the default behavior based on the type of exception? A level flag to `OpenTelemetryException?

The context is I am looking at null / arg checks which we should remove to abide by the error handling specs, that the SDK should not throw errors due to incorrect usage by a user. I think for these logging by default would be problematic since if a simple method like `.setAttribute` was logging it could easily be enough to bring down the app not so unlike an actual exception. But naturally for less common / sporadic errors like export error it would generally be ok to log.

Reply:
Since this is basically an API for error logging, I'm starting to wonder if adding this additional error handler is overkill. Couldn't users just specify a custom logger if they want special behavior, outside the normal java.util.logging behavior?

That is to say, if we stipulate the all SDK logging will be in the `io.opentelemetry.sdk` namespace, can't we just use normal logging facilities to handle the global error handler use-case?

Reply:
> The context is I am looking at null / arg checks which we should remove to abide by the error handling specs, that the SDK should not throw errors due to incorrect usage by a user.

I personally think we shouldn't log wrong parameters, if that's what you are asking. Then again, other implementations log such errors (the Python one does, for example). But we should probably discuss that in the Specification call tomorrow.

Reply:
@jkwatson What if the user wants to do something with the error, other than logging?

Reply:
Well, the logging `Handler` implementation can do whatever it wants, right? It's not required to do any actual logging, is it?

Reply:
I think I'm not following ;)

> That is to say, if we stipulate the all SDK logging will be in the io.opentelemetry.sdk namespace, can't we just use normal logging facilities to handle the global error handler use-case?

You mean entirely removing the concept of an error handler, and have user entirely rely for this purpose on `java.util.logging` services?

Reply:
exactly. @anuraaga 's comment about severity got me thinking about this. My first thought was...why not just add a severity to the global error handler. But, if we're just logging severity, message, throwable, that looks pretty much like every logging API in existence.... so why invent a new one for this purpose?

Reply:
And what *if* the user wants to do something other than simple logging? Probably we should look for cases where simple logging is not the principal use case (to go one way or another).

Reply:
by 'user', do you mean the SDK author generating the "errors" or, the user of the SDK who is doing something with them? A java.util.logging.Handler, I think, can do anything it wants...it doesn't have to be "simple logging".

Reply:
So I have mixed feelings about using the Java's logging facility. In theory having a handler is a straightforward, super simple thing to do, albeit limited, while the logging path includes dealing with an entire [LogRecord](https://docs.oracle.com/javase/7/docs/api/java/util/logging/LogRecord.html) object (and what to expect to handle there? Throwable,  message? parameters?), where the user has to override *three* abstract methods for `Handler`.

Reply:
Yeah, I'm not convinced it was the right way to go, but rather than re-invent an internal logging API, I'd rather use something off the shelf. I can also see arguments for not wanting to generate an entire stack trace just to report some sort of SDK behavior...in which case we're getting dangerously close to implementing a logging API.  :)

Reply:
Jumping in with my two-cents here… 

Taking a step back,  I think the the high-level goal here is that we want it to be easy for users of the SDK to get visibility into any errors that occur during setup or at runtime that prevent the expected tracing or metrics information from being exported. 

I think that three things that are very important for achieving this high-level goal are: 

(1) Making it easy for the SDK authors to report errors in a standardized way. 
(2) Making it easy for the SDK user (not the SDK author) to have control over where errors get logged and which types of errors get logged 
(3) Having a sane default so that important errors do not get squashed when someone is trying to set up OpenTelemetry. 

I think that both the solution outlined in this PR and using Java’s built-in logging facility technically achieve all of these goals, but I do agree with @carlosalberto that Java’s built-in logging facility potentially adds friction that makes the whole process clunkier to work with. A user has to implement three abstract methods, it’s harder to transmit information about where an error came from and switch on that error, and it’s potentially less obvious to the SDK author that calling `logger.warning` will be correctly handled and propagated to the right place. 

Having said that, I do agree with the notion that we shouldn’t be trying to implement our own logging API from scratch. If we feel like we need a lot of the bells and whistles that the Java logging library provides for our use cases, then I think that we should just go ahead and use that. If we only need a small subset of them, then I think that the overhead of building and maintaining them is worth the increased clarity and ease-of-use that this solution brings.